packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -239,14 +239,37 @@ async function redeemSharingLink(
 				const encodedShareUrl = getEncodedShareUrl(
 					odspResolvedUrl.shareLinkInfo?.sharingLinkToRedeem,
 				);
-				const redeemUrl = `${odspResolvedUrl.siteUrl}/_api/v2.0/shares/${encodedShareUrl}`;
-				const { url, headers } = getUrlAndHeadersWithAuth(
-					redeemUrl,
-					storageToken,
-					forceAccessTokenViaAuthorizationHeader,
-				);
-				headers.prefer = "redeemSharingLink";
-				return fetchAndParseAsJSONHelper(url, { headers });
+				let redeemUrl: string | undefined;
+				// There is an issue where if we use the siteUrl in /shares, then the allowed length of url is just a few hundred characters
+				// and we fail to do the redeem. But if we use the tenant domain in the url, then the allowed length becomes 2048. So, first
+				// construct the url for /shares using tenant domain but to be on safer side, fallback to what we were using before.
+				try {
```

Comment:
Do we have tests for this? It's a new flow so we should make sure this works and is tested.

Reply:
I have verified by making actual network call with tenant domain and it. works fine. Code wise it is just refactoring,

Reply:
This is a change in behavior and can regress - we are using completely different URL and have fallback mechanism in place. We should add tests that validate the new path works and that it fallbacks to the old path. This change is risky IMO.